subprojects/core/src/main/java/org/gradle/internal/buildtree/BuildTreeState.java

```diff
@@ -36,6 +36,7 @@ public class BuildTreeState implements Closeable {
 
     public BuildTreeState(ServiceRegistry parent, BuildTreeModelControllerServices.Supplier modelServices) {
         services = ServiceRegistryBuilder.builder()
+            .scope(Scopes.BuildTree.class)
```

Comment:
🤔 It seems like line 42 below should somehow provide the scope.  If my provider is the `BuildTreeScopeServices` then my scope should be `BuildTree`.  It's not good to allow the possibility of these 2 things not matching.

Reply:
It might be a good idea to enforce this statically somehow, for example where the service providers are statically typed with their scope somehow. I'd do this as a later improvement.